src/init.cpp

```diff
@@ -1315,27 +1315,27 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     // Check for host lookup allowed before parsing any network related parameters
     fNameLookup = args.GetBoolArg("-dns", DEFAULT_NAME_LOOKUP);
 
+    Proxy onion_proxy;
+
     bool proxyRandomize = args.GetBoolArg("-proxyrandomize", DEFAULT_PROXYRANDOMIZE);
     // -proxy sets a proxy for all outgoing network traffic
     // -noproxy (or -proxy=0) as well as the empty string can be used to not set a proxy, this is the default
     std::string proxyArg = args.GetArg("-proxy", "");
-    SetReachable(NET_ONION, false);
```

Comment:
The default network reachability values are implicitly set in this line in net.cpp:
```cpp
static bool vfLimited[NET_MAX] GUARDED_BY(g_maplocalhost_mutex) = {};
```

Maybe assert that each network is reachable in this first loop through them during init:

```diff
diff --git a/src/init.cpp b/src/init.cpp
index 72b86daf21..81c145870b 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -1317,6 +1317,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
         for (int n = 0; n < NET_MAX; n++) {
             enum Network net = (enum Network)n;
+            assert(IsReachable(net));
             if (!nets.count(net))
                 SetReachable(net, false);
         }
```

Reply:
Marking as resolved. I think #24205 is ok and that this PR should be merged first.

Reply:
> Marking as resolved. I think #24205 is ok and that this PR should be merged first.

Agree.